medicines/web/src/components/search-filter/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import styled from 'styled-components';
+import { DocType } from '../../services/azure-search';
+
+const StyledSearchFilter = styled.section`
+  input {
+    margin-right: 10px;
+  }
+`;
+
+interface ISearchFilterProps {
+  docTypes: DocType[];
+  checkDocType: (d: DocType) => void;
+}
+
+interface IDocTypeCheckboxProps extends ISearchFilterProps {
+  docType: DocType;
+  name: string;
+}
+
+const DocTypeCheckbox: React.FC<IDocTypeCheckboxProps> = props => {
+  const { docType: d, name, checkDocType, docTypes } = props;
```

Comment:
`docType`

Reply:
Suffers from semantic satiation because everything here is named `docType`. It's a short-lived variable. Calling it `d` in this context makes it much quicker to scan where it's coming from, than looking for `docType`.

Reply:
I agree that `docType` is better to use here - it reduces the need to understand where it's coming from at all, whereas you would always have to check where `d` comes from to work out what it is. The other doc-like variables could be renamed to make them more explicit and avoid confusion - e.g. `docTypes` could be `selectedDocTypes` or `filteredDocTypes`.

Reply:
Went for much, much more explicitly naming which should make the code read more like English now